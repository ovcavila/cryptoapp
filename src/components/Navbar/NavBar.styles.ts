import { Drawer, IconButton, SwipeableDrawer } from '@mui/material';
import styled from 'styled-components';

export const Header = styled.header`
  left: 0;
  background-color: black;
  color: white;
  display: flex;
  justify-content: space-between;
  padding: 20px 10px;
`;

export const HamburguerButton = styled(IconButton)`
  color: white;
`;

export const NavDrawer = styled(SwipeableDrawer)`
  .MuiDrawer-paper {
    width: 250px;
    background-color: black;
  }
`;
