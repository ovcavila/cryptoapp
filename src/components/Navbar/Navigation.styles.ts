import styled from 'styled-components';

export const Nav = styled.nav`
  display: flex;
  flex-direction: column;

  ${({ theme }) => theme.breakpoints.up('md')} {
    flex-direction: row;
  }
`;

export const NavLink = styled.a`
  color: white;
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  padding: 16px 8px;

  ${({ theme }) => theme.breakpoints.up('md')} {
    padding: 0 8px;
  }

  > svg {
    margin-right: 10px;
  }
`;