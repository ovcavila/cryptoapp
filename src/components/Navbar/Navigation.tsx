import { FC } from "react";
import Link from 'next/link';
import HomeIcon from '@mui/icons-material/Home';
import AutoGraphIcon from '@mui/icons-material/AutoGraph';
import PriceChangeIcon from '@mui/icons-material/PriceChange';
import ArticleIcon from '@mui/icons-material/Article';

import {Nav, NavLink} from './Navigation.styles';

export const Navigation: FC =() => (
  <Nav>
    <Link href="/" passHref>
      <NavLink>
        <HomeIcon />
        Home
      </NavLink>
    </Link>
    <Link href="/" passHref>
      <NavLink>
        <AutoGraphIcon />
        Cryptocurrencies
      </NavLink>
    </Link>
    <Link href="/" passHref>
      <NavLink>
        <PriceChangeIcon />
        Exchanges
      </NavLink>
    </Link>
    <Link href="/" passHref>
      <NavLink>
        <ArticleIcon />
        News
      </NavLink>
    </Link>
  </Nav>
);