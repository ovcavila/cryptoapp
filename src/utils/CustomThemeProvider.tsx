import { FC } from 'react';
import { ThemeProvider } from '@mui/material/styles';
import { ThemeProvider as SCThemeProvider } from 'styled-components';

import { theme } from '../../styles/theme';

export const CustomThemeProvider: FC = ({children}) => (
  <SCThemeProvider theme={theme}>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </SCThemeProvider>
);
