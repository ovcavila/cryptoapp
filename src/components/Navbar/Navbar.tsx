import { FC, useState } from "react";
import Link from 'next/link';
import { Drawer, Hidden, IconButton } from "@mui/material";

import MenuIcon from '@mui/icons-material/Menu';

import { Navigation } from "./Navigation";
import { HamburguerButton, Header, NavDrawer } from "./NavBar.styles";
import { NavLink } from "./Navigation.styles";

export const NavBar:FC = () => {
  const [openDrawer, setOpenDrawer] = useState(false);

  return (
    <>
      <Header>
        <Link href="/" passHref>
          <NavLink>CryptoApp</NavLink>
        </Link>
        <Hidden mdDown>
          <Navigation />
        </Hidden>
        <Hidden mdUp>
          <HamburguerButton onClick={() => setOpenDrawer(true)}>
            <MenuIcon />
          </HamburguerButton>
        </Hidden>
    </Header>
    <NavDrawer
      anchor="right"
      open={openDrawer}
      onOpen={() => setOpenDrawer(true)}
      onClose={() => setOpenDrawer(false)}
    >
      <Navigation />
    </NavDrawer>
  </>
  );
}