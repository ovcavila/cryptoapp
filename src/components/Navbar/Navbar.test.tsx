import React from 'react';
import { render } from '@testing-library/react';

import { CustomThemeProvider } from '../../utils/CustomThemeProvider';
import { NavBar } from './Navbar';

const setup = () => render(
  <CustomThemeProvider>
    <NavBar />
  </CustomThemeProvider>
);

describe('<Navbar />', () => {
  it('should render the navbar', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
