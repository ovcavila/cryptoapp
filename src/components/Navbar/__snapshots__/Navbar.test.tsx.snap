// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render the navbar 1`] = `
<div>
  <header
    class="sc-dkPtRN ldwmxs"
  >
    <a
      class="sc-gsDKAQ fRiYqQ"
      href="/"
    >
      CryptoApp
    </a>
  </header>
  <div
    class="PrivateSwipeArea-root css-x15wq9"
    style="width: 20px;"
  />
</div>
`;
